File: app/src/main/java/com/gystry/pjetpack/model/Comment.java
package com.gystry.pjetpack.model;

import androidx.annotation.Nullable;
import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;

import java.io.Serializable;

/**
 * @author gystry
 * 创建日期：2020/7/8 10
 * 邮箱：gystry@163.com
 * 描述：
 */
public class Comment extends BaseObservable implements Serializable {
    public int id;
    public long itemId;
    public long commentId;
    public long userId;
    public int commentType;
    public long createTime;
    public int commentCount;
    public int likeCount;
    public String commentText;
    public String imageUrl;
    public String videoUrl;
    public int width;
    public int height;
    public boolean hasLiked;
    public User author;
    public Ugc ugc;

    @Bindable
    public Ugc getUgc() {
        if (ugc == null) {
            ugc = new Ugc();
        }
        return ugc;
    }

    public void setUgc(Ugc ugc) {
        this.ugc = ugc;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj == null || !(obj instanceof Comment))
            return false;

        Comment newComment = (Comment) obj;
        return likeCount == newComment.likeCount
                && hasLiked == newComment.hasLiked
                && (author != null && author.equals(newComment.author))
                && (ugc != null && ugc.equals(newComment.ugc));
    }

}

File: appkt/src/main/java/com/gystry/appkt/model/Comment.kt
package com.gystry.appkt.model

import java.io.Serializable

/**
 * @author gystry
 * 创建日期：2021/3/30 11
 * 邮箱：gystry@163.com
 * 描述：
 */
data class Comment(var id: Int = 0,
                   var itemId: Long = 0,
                   var commentId: Long = 0,
                   var userId: Long = 0,
                   var commentType: Int = 0,
                   var createTime: Long = 0,
                   var commentCount: Int = 0,
                   var likeCount: Int = 0,
                   var commentText: String,
                   var imageUrl: String,
                   var videoUrl: String,
                   var width: Int = 0,
                   var height: Int = 0,
                   var hasLiked: Boolean = false,
                   var author: User,
                   var ugc: Ugc,): Serializable {
}